app/src/main/java/com/brainwallet/data/repository/LtcRepository.kt

```diff
@@ -47,33 +55,50 @@ interface LtcRepository {
         }
 
         override suspend fun fetchFeePerKb(): Fee {
-            val lastUpdateTime = sharedPreferences.getLong(PREF_KEY_NETWORK_FEE_PER_KB_CACHED_AT, 0)
-            val currentTime = System.currentTimeMillis()
-            val cachedFee = sharedPreferences.getString(PREF_KEY_NETWORK_FEE_PER_KB, null)
-                ?.let { json.decodeFromString<Fee>(it) }
+            return sharedPreferences.fetchWithCache(
+                key = PREF_KEY_NETWORK_FEE_PER_KB,
+                cachedAtKey = PREF_KEY_NETWORK_FEE_PER_KB_CACHED_AT,
+                cacheTimeMs = 6 * 60 * 60 * 1000,
+                fetchData = {
+                    remoteApiSource.getFeePerKb()
+                },
+                defaultValue = Fee.Default
+            )
+        }
 
-            return runCatching {
-                // Check if cache exists and is less than 6 hours old
-                if (cachedFee != null && (currentTime - lastUpdateTime) < 6 * 60 * 60 * 1000) {
-                    return cachedFee
+        override suspend fun fetchLimits(baseCurrencyCode: String): MoonpayCurrencyLimit {
+            return sharedPreferences.fetchWithCache(
+                key = "${PREF_KEY_BUY_LIMITS_PREFIX}${baseCurrencyCode.lowercase()}",
+                cachedAtKey = "${PREF_KEY_BUY_LIMITS_PREFIX_CACHED_AT}${baseCurrencyCode.lowercase()}",
+                cacheTimeMs = 5 * 60 * 1000, //5 minutes
+                fetchData = {
+                    remoteApiSource.getMoonpayCurrencyLimit(baseCurrencyCode)
                 }
+            )
+        }
 
-                val fee = remoteApiSource.getFeePerKb()
-                sharedPreferences.edit {
-                    putString(PREF_KEY_NETWORK_FEE_PER_KB, json.encodeToString(fee))
-                    putLong(PREF_KEY_NETWORK_FEE_PER_KB_CACHED_AT, currentTime)
-                }
+        override suspend fun fetchBuyQuote(params: Map<String, String>): GetMoonpayBuyQuoteResponse =
+            remoteApiSource.getBuyQuote(params)
 
-                return fee
-            }.getOrElse {
-                cachedFee ?: Fee.Default
-            }
+        override suspend fun fetchMoonpaySignedUrl(params: Map<String, String>): String {
+            return remoteApiSource.getMoonpaySignedUrl(params)
+                .signedUrl.toUri()
+                .buildUpon()
+                .apply {
+                    if (BuildConfig.DEBUG) {
+                        authority("buy-sandbox.moonpay.com")//replace base url from buy.moonpay.com
```

Comment:
when I am trying to replace the base url for sandbox, found warning like this and somehow can't load the webview

| using original baseurl | replaced for sandbox base url |
|----------------------|-------------------------------|
|  ![moonpayprod](https://github.com/user-attachments/assets/4ce5e28e-1cba-4697-8e11-d392b91d8b2d) |  ![moonpaysandbox](https://github.com/user-attachments/assets/370f7595-bb4c-4e97-be49-2a73aa6369c3) |
| buy.moonpay.com | buy-sandbox.moonpay.com |


and this is log when I debug the webview
<img width="497" alt="Screenshot 2025-05-10 at 15 28 27" src="https://github.com/user-attachments/assets/649d035a-2ac2-4a8b-a479-4316307b35ec" />

cc @kcw-grunt @josikie

Reply:
Thanks @andhikayuana . Let me take a look. 👀

Reply:
We are very close @andhikayuana .  The biggest blocking issue is the portal to the MoonPay must be as big as possible . Right now it looks tiny and that will cause people to quit or complain.

Please take these steps:

1. Reduce the padding in the header to as little as possible
2. Widen the widow ams much as possible so that it still floats over the main app view but clearly a separate focus
3. Lengthen the portal window and much as possible and maybe trigger it to lengthen when they proceed to tap Buy after qty is picked

![Screenshot 2025-05-11 at 10 32 50](https://github.com/user-attachments/assets/cb14c068-a3f7-49ad-b32f-3d2f5064a8df)

Reply:
Let me look at the js issue you are experiencing

Reply:
@andhikayuana  sent a slack